### PR TITLE
refactor: package and package sidebar code refactoring

### PR DESCRIPTION
**Pull request by ayusharma**
**Type:**

The following has been addressed in the PR:

*  There is a related issue #829 #780 
*  Unit or Functional tests are included in the PR. 

**Description:**
I have done heavy refactoring. The aim is to have more functional component instead of components which handles a lot of complexity. Currently `<Infos />, <Dependencies /> <LastSync />` etc.. components have many methods to for `author`, `repository`, `license` and this PR tries to clean that and put all things from a central place. 

I have added a `package` file in utils which formats the differents field of `pakage json` and pass to functional components.

### CODE REVIEW

File: src/webui/components/Package/index.js [hunk 1]
```diff
@@ -2,81 +2,51 @@ import React from 'react';
...
-  }
-
-  renderTitle(pkg) {
-    return (
-      <div className={classes.title}>
-        <h1>
-          {pkg.name} {this.renderTag(pkg)}
-        </h1>
+const Package = ({name, version, author, description, license, time}) => {
+  console.log(name, version, author, description, license, time);
```
Comment (juanpicado) [inline src/webui/components/Package/index.js]: `console.log()`

Comment (juanpicado) [reply src/webui/components/Package/index.js]: Please enable strict rules for UI, any console.log should pass the linting

Comment (ayusharma) [reply src/webui/components/Package/index.js]: I will push that change soon.

File: src/webui/components/PackageSidebar/modules/Maintainers/index.jsx [hunk 2]
```diff
@@ -44,7 +47,10 @@ export default class Maintainers extends React.Component {
       return [];
     }
 
-    return uniqBy(this.contributors, (contributor) => contributor.name).slice(0, 5);
+    return uniqBy(this.contributors, (contributor) => contributor.name).slice(
+      0,
+      5
```
Comment (juanpicado) [inline src/webui/components/PackageSidebar/modules/Maintainers/index.jsx]: Convert 5 to const above, then all know what that means.

Comment (juanpicado) [reply src/webui/components/PackageSidebar/modules/Maintainers/index.jsx]: Even we can use that value as a future config on the web object.

File: src/webui/components/PackageSidebar/modules/Maintainers/index.jsx [hunk 3]
```diff
@@ -56,26 +62,33 @@ export default class Maintainers extends React.Component {
...
+          avatar={contributor.avatar}
+        />
+      );
+    });
   }
 
-  render() {
-    let author = this.author;
-
+  rednerAuthorAndContributors(author) {
```
Comment (juanpicado) [inline src/webui/components/PackageSidebar/modules/Maintainers/index.jsx]: typo -> `render`

File: test/unit/empty.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export default {};
```
Comment (juanpicado) [inline test/unit/empty.js]: ????? 🤔

Comment (juanpicado) [reply test/unit/empty.js]: I've had never seen such performance improvement, brilliant. 😆

Comment (ayusharma) [reply test/unit/empty.js]: It is for `svg`. https://github.com/verdaccio/verdaccio/blob/4a6383d05f90814c081bcab9e229638dc51f7bfc/jest.config.js#L45 I tried `svg` with identity-obj-proxy but facing some errors.

Comment (juanpicado) [reply test/unit/empty.js]: How is the trick?

Comment (ayusharma) [reply test/unit/empty.js]: ` '\\.(svg)$': '<rootDir>/test/unit/empty.js'` in jest.config.js

**Pull request merged**